### PR TITLE
Issue 12327: Convert avatars to static

### LINKED ISSUE #12327: [mastodon API] disabling GIF avatars in Fedilab not functional since Friendica does not differentiate between animated and static picture
### Bug Description

Fedilabs option to disable GIF avatars does not disable GIF avatars.

As per https://codeberg.org/tom79/Fedilab/issues/428 it seems Friendica is not correctly differentiating between the two paths used for animated and static picture.

### Steps to Reproduce

1.  for Friendica account in Fedilab open Settings > Timelines > and enable `Disable GIF avatars` option

Actual Result:

GIF avatars still animated.

Expected Result:

GIF avatars showing as static image

### Platform Info

Friendica Version: This is Friendica, version 2022.12-dev that is running at the web location https://pirati.ca. The database version is 1497/1497, the post update version is 1484/1484.
Fedilab 3.9.3

### PR DESCRIPTION
**Pull request by annando**
Fixes #12327

### CODE REVIEW

File: src/Model/Contact.php [hunk 1]
```diff
@@ -2057,9 +2057,10 @@ public static function getDefaultAvatar(array $contact, string $size): string
 	 * @param integer $cid     contact id
 	 * @param string  $size    One of the Proxy::SIZE_* constants
 	 * @param string  $updated Contact update date
+	 * @param bool    $static  If "true" a parameter is added to convert the header to a static one
```
Comment (MrPetovan) [inline src/Model/Contact.php]: "header"?

File: src/Module/Photo.php [hunk 2]
```diff
@@ -182,6 +182,12 @@ protected function rawContent(array $request = [])
 			throw new HTTPException\InternalServerErrorException($error);
 		}
 
+		if (!empty($request['static'])) {
+			$img = new Image($imgdata, $photo['type']);
+			$img->toStatic();
+			$imgdata = $img->asString();
```
Comment (MrPetovan) [inline src/Module/Photo.php:188]: This should be cached.

File: src/Object/Image.php [hunk 3]
```diff
@@ -57,7 +57,7 @@ class Image
 	 */
 	public function __construct(string $data, string $type = null)
 	{
-		$this->imagick = class_exists('Imagick');
+		$this->imagick = class_exists('Imagick') && !class_exists('GDImage');
```
Comment (MrPetovan) [inline src/Object/Image.php]: Why are we preferring gd to Imagemagick?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Contact.php (hunk 1 above)
Comment (annando) [reply src/Model/Contact.php]: Fixed

File: src/Module/Photo.php (hunk 2 above)
Comment (annando) [reply src/Module/Photo.php:188]: Caching is not needed here, since we are sending the appropriate headers, so that the browser will perform the caching.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**